ledger/src/blockstore.rs

```diff
@@ -2649,6 +2649,21 @@ impl Blockstore {
         matches!(self.db.get::<cf::Root>(slot), Ok(Some(true)))
     }
 
+    /// Returns true if a slot is between the rooted slot bounds of the ledger, but has not itself
+    /// been rooted. This is either because the slot was skipped, or due to a gap in ledger data,
+    /// as when booting from a newer snapshot.
+    pub fn is_skipped(&self, slot: Slot) -> bool {
+        let lowest_root = self
+            .rooted_slot_iterator(0)
+            .ok()
+            .and_then(|mut iter| iter.next())
+            .unwrap_or_default();
+        match self.db.get::<cf::Root>(slot).ok().flatten() {
+            Some(_) => false,
+            None => slot < self.max_root() && slot > lowest_root,
```

Comment:
If `slot < lowest_root` does that necessarily mean the slot was cleaned up?

Thinking of an edge case like:

1) Lowest root is 5
2) We check confirmation on slot 6, finds null
3) We call `is_skipped`
4) Cleanup cleans up to slot 7
5) `is_skipped` finds slot < lowest_root, but returns false

Reply:
If you have a suggestion for handling this better, I'm open to it. But I don't think we mind this case too much, since it will be caught lower in the method and return slot cleaned up. The prime use case here is near the max-root.
I do need to think about whether we want to return this error for skipped slots in the range of BigTable, though...